tests/unit/operation/relate/RelateOpTest.cpp

```diff
@@ -0,0 +1,79 @@
+//
+// Test Suite for geos::operation::relate::RelateOp class
+
+#include <tut/tut.hpp>
+// geos
+#include <geos/constants.h> // for std::isnan
+#include <geos/geom/Coordinate.h>
+#include <geos/geom/CoordinateArraySequence.h>
+#include <geos/geom/Dimension.h>
+#include <geos/geom/Geometry.h>
+#include <geos/geom/LineString.h>
+#include <geos/geom/GeometryFactory.h>
+#include <geos/geom/PrecisionModel.h>
+#include <geos/io/WKTReader.h>
+#include <geos/operation/relate/RelateOp.h>
+
+// std
+#include <cmath>
+#include <string>
+#include <memory>
+
+using namespace geos::geom;
+using namespace geos::operation::relate;
+
+namespace tut {
+//
+// Test Group
+//
+
+struct test_relateop_data {
+    geos::io::WKTReader wktreader;
+    typedef std::unique_ptr<Geometry> GeomPtr;
+    typedef geos::geom::GeometryFactory GeometryFactory;
+
+    geos::geom::PrecisionModel pm_;
+    GeometryFactory::Ptr factory_;
+
+    test_relateop_data()
+        : pm_(1), factory_(GeometryFactory::create(&pm_, 0))
+    {}
+
+    void checkRelate(const char* wkta, const char* wktb, const char* imExpected)
```

Comment:
you can declare this `checkRelate(const std::string& wkta, const std::string& wktb, const std::string& imExpected)` and skip the extra string vars below

Reply:
Nice.  So many wonderlful choices in C++!